src/sentry/api/serializers/models/doc_integration.py

```diff
@@ -0,0 +1,39 @@
+from typing import Any, Mapping
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models.integration import DocIntegration
+from sentry.models.integrationfeature import IntegrationFeature, IntegrationTypes
+from sentry.models.user import User
+from sentry.utils.compat import map
+from sentry.utils.json import JSONData
+
+
+@register(DocIntegration)
+class DocIntegrationSerializer(Serializer):
+    def serialize(
+        self,
+        obj: DocIntegration,
+        attrs: Mapping[str, Any],
+        user: User,
+        **kwargs: Any,
+    ) -> JSONData:
+        from sentry.api.validators.doc_integration import METADATA_PROPERTIES
+
+        data = {
+            "name": obj.name,
+            "author": obj.author,
+            "description": obj.description,
+            "url": obj.url,
+            "popularity": obj.popularity,
+            "isDraft": obj.is_draft,
+        }
+
+        if obj.metadata:
+            for metadata_property in METADATA_PROPERTIES:
+                data[metadata_property] = obj.metadata.get(metadata_property)
+
+        features = IntegrationFeature.objects.filter(
```

Comment:
You should do this DB query in the `get_attrs()` call. It'd look something like:
```py
def get_attrs(self, item_list, user, **kwargs):
    target_ids = set(item.id for item in item_list)
    features = IntegrationFeature.objects.filter(
        target_type=IntegrationTypes.DOC_INTEGRATION.value,
        target_id__in=target_ids,
    )

    features_by_target = defaultdict(set)
    for feature in features:
        features_by_target[feature.target_id].add(feature)

    return {
        item: {"features": features_by_target.get(item.id, set())}
        for item in item_list
    }
```
The idea is that you should be able to call `serialize()` without any DB side-effects.
```py
def serialize(
    self,
    obj: DocIntegration,
    attrs: Mapping[str, Any],
    user: User,
    **kwargs: Any,
) -> JSONData:
    data = {
        ...
        "isDraft": obj.is_draft,
        "features": map(lambda x: serialize(x, user), attrs.get("features"))
    }